### DISCUSSION TITLE
Welcome to wrp Discussions!

### DISCUSSION THREAD

**Discussion by tenox7** [tenox7/wrp#82] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by retrofan2006** [comment]
Can you add another recapturing solution? Because the St button in WRP stops the webpage from loading. You have to add a separate recapture button that doesn't stops the loading, just makes a screenshot about the current state of the webpage, just like in Eric Mackrodt's solution. Also you have to add a JPEG option and quality control (from 100 to 1000) for faster rendering. By the way, WRP is a very nice software. I've tried it on old feature phone and smartphone browsers that uses client side webpage rendering, and it worked pretty well. Just add those two things in the next version that I've mentioned to make it even better. That would be really awesome.

**Reply by tenox7** [reply]
Thank you for opening a discussion for this. I saw your comment on the blog about performance. Could you open individual bugs under Issues for all things you want added? Like performance, recapture, jpeg option, etc. It would be nice to have everything in one place. Thanks!

**Comment by Wayfinlay** [comment]
I'm running WRP on my windows 11 machine and trying to connect on my windows 98 computer as a client of course. I've tried everything for weeks. How do I know what IP Address goes in the URL? Begging for somebody's help. Thank you in advance!

**Reply by tenox7** [reply]
In Windows command prompt type: `ipconfig` this will show you the IP address. If running PowerShell use `Get-NetIPAddress` instead. I would also recommend to disable Windows firewall or make an exception for WRP.

**Reply by Wayfinlay** [reply]
You're so legendary! Thank you so much. Gonna test it

**Reply by Wayfinlay** [reply]
Why do you think this doesn't work? I'm doing everything correct, right?
![thumbnail](https://user-images.githubusercontent.com/118699750/206168051-07d8d0cf-1716-40ee-ba95-29853ef9b732.jpg)
![thumbnail](https://user-images.githubusercontent.com/118699750/206168061-36348b81-67ed-4c0a-b954-e421519e3661.jpg)
![thumbnail](https://user-images.githubusercontent.com/118699750/206168075-521e3086-c9b0-43f9-b848-ab4e5dde191c.jpg)
![thumbnail](https://user-images.githubusercontent.com/118699750/206168091-37147d8a-6dfb-4a46-861f-7c0f010ccef6.jpg)

**Reply by tenox7** [reply]
Looking at the screenshots, I think you are changing IP address of the Windows 98 machine to the same as Windows 11? 

* Keep both machines on DHCP, do not manually set any IP addresses
* On Windows 11 in the cmd window, before you run wrp type ipconfig and get your IP address
* On Windows 98 open the web browser and in the web browser enter: http://x.x.x.x:8080 where x.x.x.x is IP address of your windows 11 machine - presumably 192.168.1.23

**Reply by Wayfinlay** [reply]
I am very appreciative of your help. For some reason I just can't get it to work. I put them both on DHCP like you said but same result. I'm supposed to have the two computers connected through ethernet cable right? Should I connect them before or after starting the server? If any of that matters

**Reply by tenox7** [reply]
You should connect them before starting wrp server. Also before you start try `ping` from win98 to win11. To run ping, on win98 machine type `ping x.x.x.x` where x.x.x.x is the IP address of Windows 11 machine.

**Reply by Wayfinlay** [reply]
Ok cool! Last question. Where on the windows 98 machine do I type in the ping?

**Reply by tenox7** [reply]
you have to find "Command Prompt" somewhere in the Start menu, or perhaps type "command" in Run...

**Reply by Wayfinlay** [reply]
Sorry to bother you again. Still going at it :(. So I start the server and it starts just like normal on my windows 11 machine. I plug in an ethernet cable into both computers together. I turn on the windows 98 machine and I typed the "ping x.x.x.x" using the IP address obtained from the windows 11 command prompt ipconfig. It says "request timed out". Then if I type in the ip address from the windows 11 machine, it says " page cannot be displayed. Did I miss anything? Are you sure I don't need to change any ip address settings on the win 98 machine? Much appreciated.

**Reply by tenox7** [reply]
Not quite. The computers should not be connected by an ethernet cable to each other. They should be connected to the same switch together with your router, etc. They should be able to go to internet independently. Also you don't need to start wrp to run ping command.

**Reply by Wayfinlay** [reply]
Thank you for baring with me. Sounds good. I'll try it tomorrow

**Comment by ghost** [comment]
Hi

im terribly sorry if this is a double post (also asked here: https://github.com/tenox7/wrp/issues/111)

This might be far fetched, but i'll ask it anyways. I'm trying to run this on my NAS (QNAP) on the container station (docker) and i'm fetching/starting it with a regular:

docker run -d -p 85:8085 tenox7/wrp

as per the instructions (as both these ports are free on my environment). When looking at the logs, i see this:

"2023/10/02 19:19:18 Web Rendering Proxy Version 4.6.0
2023/10/02 19:19:18 Args: ["/wrp"]
2023/10/02 19:19:18 Default Img Type: gif, Geometry: {w:1152 h:600 c:216}
2023/10/02 19:19:18 Got UI template from built-in
2023/10/02 19:19:18 Starting WRP http server on :8080"

(i.e the instruction to bind to external port 8085 does not catch on)

It seems to be binding to 8080 (a port currently occupied by another process on this device). What can i do to circument this?

I must also say that i did download it for the Win environment (10) and it works perfectly fine using the default 8080 exposed port on there (absolutely perfect as a matter of fact).

What i try to achieve is basically to point my extremely vintage hardware (Atari STe) to this proxy in order to render some more modern pages on it :)

**Comment by ghost** [comment]
Hello,

I have just started it again 10 times (no firewall, with administrator) but it still wont work.
It keeps stuck at:

_**Starting WRP http server**_

This is very annoying. How do i fix this?
Greetings, Jensvp2

**Reply by tenox7** [reply]
Can you let me know what OS is this on?

**Reply by ghost** [reply]
Microsoft Windows [Version 10.0.19045.4170]

**Reply by tenox7** [reply]
and is it just sitting on "starting wrp http server" ?

**Reply by ghost** [reply]
yes, tried multiple times with the same output. i had it run in the night, but didnt continue further.

**Reply by tenox7** [reply]
The message "Starting HTTP Server" is the LAST startup message. There is nothing further beyond this. It means that the server successfully started. You should be able to connect tot he specified port with a web browser. Port 8080 by default.

**Reply by ghost** [reply]
I have tried, but my DSI didnt connect to the proxy. I tried every IP it gave.

**Reply by tenox7** [reply]
Please do not use WRP as a proxy. Just point your browser to it's IP address. Start by pointing a local web browser on Windows to http://localhost:8080/

**Reply by ghost** [reply]
When i go to http:// (IP adress) it refuses to connect. Even when i put :8080 after that. :/
